### PR TITLE
Fix GitLab Trusted Publisher not accepting valid namespaces

### PR DESCRIPTION
**Pull request by facutuesca**
Fixes https://github.com/pypi/warehouse/issues/15836.

GitLab namespaces can have forward slash ("/") characters, since a repo inside a subgroup will have a namespace where the subgroups are delimited using slashes (e.g: `gitlab.com/group/subgroup/repo` has a namespace of `group/subgroup`).

I double checked by generating an OIDC token from a GitLab CI workflow in a repo inside a subgroup, the namespace claim is of the form `group/subgroup`.

cc @di @woodruffw

### CODE REVIEW

File: warehouse/oidc/forms/gitlab.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 # https://docs.gitlab.com/ee/user/reserved_names.html#limitations-on-project-and-group-names
 _VALID_GITLAB_PROJECT = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-_.]*$")
-_VALID_GITLAB_NAMESPACE = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-_.]*$")
+_VALID_GITLAB_NAMESPACE = re.compile(r"^[a-zA-Z0-9][a-zA-Z0-9-_./]*$")
```
Comment (di) [inline warehouse/oidc/forms/gitlab.py:23]: This permits trailing slashes `foobar/` and multiple repeated slashes `foo//bar`, both of which I think should be invalid.

Comment (facutuesca) [reply warehouse/oidc/forms/gitlab.py:23]: That is also the case also for the other special symbols ([src](https://docs.gitlab.com/ee/user/reserved_names.html#limitations-on-usernames-project-and-group-names)):
> Usernames, project or group slugs:
> - Cannot start or end with a special character.
> - Must not contain consecutive special characters.

And there are also other limitations we don't check (like the project ending with `.git` or `.atom`. At the time, the reasoning was that we wanted some validation for the user input that discarded simple errors but didn't need to be exhaustive about rejecting invalid input (since we're not hitting the GitLab API to validate them).

Should we have more complex regexes and/or validation functions to account for all of the possible invalid inputs?

Comment (di) [reply warehouse/oidc/forms/gitlab.py:23]: Let's merge this as-is and create a new issue to make the regexes a little more accurate.

File: tests/unit/oidc/forms/test_gitlab.py [hunk 2]
```diff
@@ -76,6 +84,11 @@ def test_validate(self):
                 "project": "some",
                 "workflow_filepath": "some",
             },
+            {
+                "namespace": "/start_with_slash",
```
Comment (di) [inline tests/unit/oidc/forms/test_gitlab.py:88]: Is this a valid namespace? Or is this an edge case we shouldn't actually permit?

Comment (woodruffw) [reply tests/unit/oidc/forms/test_gitlab.py:88]: From GitLab's docs, I believe this is invalid:

> Usernames, project and group names must start and end with a letter (a-zA-Z) or a digit (0-9)

Comment (facutuesca) [reply tests/unit/oidc/forms/test_gitlab.py:88]: it's invalid (it's part of the test that checks invalid namespaces)

Comment (woodruffw) [reply tests/unit/oidc/forms/test_gitlab.py:88]: Whoops, missed that!

**Review by woodruffw — APPROVED** [review]
Thanks @facutuesca!

File: tests/unit/oidc/forms/test_gitlab.py (hunk 2 above)
Comment (di) [reply tests/unit/oidc/forms/test_gitlab.py:88]: Sorry, also missed that!

**Review by di — APPROVED** [review]

File: warehouse/oidc/forms/gitlab.py (hunk 1 above)
Comment (facutuesca) [reply warehouse/oidc/forms/gitlab.py:23]: issue created here: https://github.com/pypi/warehouse/issues/15852
